Issue with multiform data rest client on Quarkus 3.14

### Describe the bug

I have the below rest endpoint. Whenever I start Quarkus, I get an error saying that my rest client has not been indexed at build time. I am using Quarkus 3.14.1

```java
public class Request {
    
    @RestForm("file")
    @PartType(MediaType.APPLICATION_OCTET_STREAM)
    private InputStream file;

    @RestForm("filename")
    @PartType(MediaType.TEXT_PLAIN)
    private String filename;
}
```

Endpoint definition:

```java
    @POST
    @Path("/upload")
    @Consumes(MediaType.MULTIPART_FORM_DATA)
    @Produces(MediaType.APPLICATION_JSON)
    public String upload(Request request) {
      // business logic.
        return "done";
    }
```

Error: 
Caused by: java.lang.IllegalStateException: REST client interface: interface com.test.api.restclient.TestRestApi was not indexed at build time.


### Expected behavior

The expectation is that the app should start up without any issues. 

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

I have created 2 sample projects. In my consumer project, I run a test inside ExampleResourceTest.java and that is where I get the error message.

Error: 
Caused by: java.lang.IllegalStateException: REST client interface: interface za.co.tutorial.StickerDefinitionRestApi was not indexed at build time. See https://quarkus.io/guides/cdi-reference#bean_discovery for information on how to index the module that contains it.
	at org.jboss.resteasy.reactive.client.impl.ClientProxies.get(ClientProxies.java:35)
	at org.jboss.resteasy.reactive.client.impl.WebTargetImpl.proxy(WebTargetImpl.java:449)
	at io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl.build(RestClientBuilderImpl.java:528)
[consumer.zip](https://github.com/user-attachments/files/17089308/consumer.zip)

The producer is greater than 25MB so I will find a way to send you a link to my repo.

--- a/extensions/resteasy-reactive/rest-client-jaxrs/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java
+++ b/extensions/resteasy-reactive/rest-client-jaxrs/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java
@@ -1964,7 +1964,7 @@ public class JaxrsClientReactiveProcessor {
                 methodCreator.invokeVirtualMethod(MethodDescriptor.ofMethod(ClientMultipartForm.class, "entity",
                         ClientMultipartForm.class, String.class, String.class, Object.class, String.class, Class.class),
                         multipartForm, formParamResult, partFilenameResult, fieldValue,
-                        methodCreator.load(partType),
+                        methodCreator.load(partType != null ? partType : MediaType.APPLICATION_OCTET_STREAM),
                         // FIXME: doesn't support generics
                         methodCreator.loadClassFromTCCL(type)));
     }

--- a/extensions/resteasy-reactive/rest-client/deployment/src/test/java/io/quarkus/rest/client/reactive/multipart/MultipartFilenameTest.java
+++ b/extensions/resteasy-reactive/rest-client/deployment/src/test/java/io/quarkus/rest/client/reactive/multipart/MultipartFilenameTest.java
@@ -401,7 +401,6 @@ public class MultipartFilenameTest {
 
     public static class ClientFormUsingInputStream {
         @FormParam("myFile")
-        @PartType(APPLICATION_OCTET_STREAM)
         @PartFilename(FILE_NAME)
         public InputStream file;
     }

